optimum/intel/ipex/inference.py

```diff
@@ -0,0 +1,78 @@
+import intel_extension_for_pytorch as ipex
+import torch
+from torch import nn
+from transformers.pipelines import Pipeline
+from typing import Union
+
+
+class _ModelFallbackWrapper:
+
+    __slots__ = ("_optimized", "_default")
+
+    def __init__(self, optimized, default):
+        self._optimized = optimized
+        self._default = default
+
+    def __call__(self, *args, **kwargs):
+        try:
+            return self._optimized(*args, **kwargs)
+        except Exception:
+            return self._default(*args, **kwargs)
+
+    def __getattr__(self, item):
+        if not item.startswith("__"):
+            return getattr(self._default, item)
+        else:
+            return self.item
+
+
+class inference_mode:
+    __slots__ = ("_model", "_dtype", "_graph_mode", "_use_onednn", "_verbose", "_original")
+
+    def __init__(
+        self,
+        model: Union[nn.Module, Pipeline],
+        dtype: torch.dtype = torch.float32,
+        use_onednn: bool = True,
+        verbose: bool = False

```

Comment:
It could be nice to add a docstring to give more information on these parameters

Reply:
Done ✔️